src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -128,14 +128,15 @@ protected override void ProcessRecord()
             {
                 byte[] bytehash = null;
                 string hash = null;
-                Stream openfilestream = null;
 
                 try
                 {
-                    openfilestream = File.OpenRead(path);
-                    bytehash = hasher.ComputeHash(openfilestream);
-
-                    hash = BitConverter.ToString(bytehash).Replace("-", string.Empty);
+                    using (Stream openfilestream = File.OpenRead(path))
+                    {
+                        bytehash = hasher.ComputeHash(openfilestream);
+                        hash = BitConverter.ToString(bytehash).Replace("-", string.Empty);
+                    }
+                    
                     WriteHashResult(Algorithm, hash, path);
```

Comment:
We can not use `using` and put the write in`finally` like 
```c#
if (hash != null)
{                
    WriteHashResult(Algorithm, hash, path);
}
```

Reply:
We can, sure. Is there much of a reason to use that pattern? It requires an additional check which we don't really have a need to do.

Reply:
Is `using` something to avoid? I like how `using` keeps the scope for the open file together in just a few lines, but your suggestion would be fine as well - I can change it?

Or add another `openfilestream?.Dispose()` just before WriteHashResult, and keep the one in `finally` as well in case of exceptions?

Reply:
`using` is internally try-finally. Really it masks logic error. If we want to add something in the future, the proposed separation of logic will save us from possible problems like this.

Reply:
What logic error are you seeing here? What'd I miss? 😄

Reply:
Ideally it could be:
```c#
if (ComputeFileHash(Algorithm, path, out hash)
{
    WriteHashResult(Algorithm, hash, path);
}

private static bool ComputeFileHash(...)
{
    try
    {
    }
    catch
    {
    }
    finally
    {
    }
}

Reply:
> What logic error

This is the bug we fix here - we should dispose before write.

Reply:
That'll have the same issue, though.

This problem arises because we're not disposing the file handle before we write the result. That's what we need to do here, and separating them out like that will bring the problem back. 

We need to:

1. Open file
2. Read file
3. Compute hash
4. Close file handle
5. Write hash result

In that order, or we come back to the same problem here.

Reply:
In general, nested try-try (that `using` is internally) should be avoided until it is inevitable. This complicates the reading and understanding of the code. My second example is perfectly simple from this point of view.

Reply:
OK, code rewritten with dedicated ComputeHash method.